tests/Tests.Json.Decode.fs

```diff
@@ -2158,54 +2150,40 @@ Expecting an object with a field named `radius` but instead got:
                           Must = "must value" } : RecordWithOptionalClass)
                 equal expected actual
 
-            testCase "Auto.fromString returns an Error for field using a non optional class" <| fun _ ->
-                let json = """{ "class" : null, "must": "must value"}"""
-                let actual = Decode.Auto.fromString<RecordWithRequiredClass>(json, isCamelCase=true)
-                #if FABLE_COMPILER
-                let expected =
-                    Error (
-                        """
-Error at: `$.class`
-I run into a `fail` decoder: Class types cannot be automatically deserialized: Tests.Decode.BaseClass
-                        """.Trim())
-                #else
+            testCase "Auto.fromString works for records missing optional field value on classes" <| fun _ ->
+                let json = """{ "must": "must value"}"""
+                let actual = Decode.Auto.fromString<RecordWithOptionalClass>(json, isCamelCase=true)
                 let expected =
-                    Error (
-                        """
-Error at: `$.class`
-I run into a `fail` decoder: Class types cannot be automatically deserialized: Tests.Decode+BaseClass
-                        """.Trim())
-                #endif
+                    Ok ({ MaybeClass = None
+                          Must = "must value" } : RecordWithOptionalClass)
                 equal expected actual
 
+            testCase "Auto.generateDecoder throws for field using a non optional class" <| fun _ ->
+                let expected = "Cannot generate auto decoder for Tests.Decode.BaseClass. Please pass an extra decoder."
+                let errorMsg =
+                    try
+                        let decoder = Decode.Auto.generateDecoder<RecordWithRequiredClass>(isCamelCase=true)
+                        ""
+                    with ex ->
+                        ex.Message
+                errorMsg.Replace("+", ".") |> equal expected
```

Comment:
Why do we need this `Replace` call ?

Reply:
In .NET reflection `.` is used for namespaces and `+` for inner classes (modules are compiled as static classes so classes within a module are inner classes), so the error in .NET actually reads: "Cannot generate auto decoder for Tests.Decode+BaseClass". But Fable just uses `.` for everything.